app/questionnaire/placeholder_transforms.py

```diff
@@ -80,11 +81,27 @@ def format_number(number):
 
         return ''
 
-    def calculate_years_difference(self, first_date, second_date):
-        first_date = datetime.now() if first_date == 'now' else datetime.strptime(first_date, self.input_date_format)
-        second_date = datetime.now() if second_date == 'now' else datetime.strptime(second_date, self.input_date_format)
+    @staticmethod
+    def calculate_years_difference(first_date, second_date):
+        return str(relativedelta(PlaceholderTransforms.parse_date(second_date),
+                                 PlaceholderTransforms.parse_date(first_date)).years)
+
+    @staticmethod
+    def parse_date(date):
+        """
+        :param date: string representing a date
+        :return: datetime of that date
+
+        Convert `date` from string into `datetime` object. `date` can be 'YYYY-MM-DD', 'YYYY-MM'
+        or 'now'. Note that in the shorthand YYYY-MM format, day_of_month is assumed to be 1.
+        """
+        if date == 'now':
+            return datetime.now()
```

Comment:
Should this be UTC?

Reply:
I didn't want to change the behaviour any more than necessary to fix the bug, but if you chaps would like to do that, I can, certainly.

Reply:
This should be UTC AFAICS. I can't see any reason to work with local time on the server side.

It's likely never going to cause a major issue since we're not working with times, but probably worth cleaning up while here.

Reply:
Hmm, given that it is date only, how does TZ come into play? And given that we are not aware of the TZ of the other date provided, it wouldn't be enough to change the outcome even if we were asking for the time.

Reply:
`datetime.now()` returns a local datetime, so if it's around midnight, the timezone can cause a change to the date.

Agree that the timezone of the other date is not defined but given that this is a generic method, I think it should assume UTC to keep things consistent rather than using local time which can vary depending on the server this runs on.